config/config.go

```diff
@@ -65,34 +70,95 @@ func ReadConfig(configFilePath string) (*Configuration, *common.Block, error) {
 			return nil, nil, errors.Wrapf(err, "failed to read shared config, path is empty")
 		}
 	case "block":
-		if conf.LocalConfig.NodeLocalConfig.GeneralConfig.Bootstrap.File != "" {
-			blockPath := conf.LocalConfig.NodeLocalConfig.GeneralConfig.Bootstrap.File
-			data, err := os.ReadFile(conf.LocalConfig.NodeLocalConfig.GeneralConfig.Bootstrap.File)
+		// If node is router or batcher, check if config store has blocks, and if yes bootstrap from the last block
+		if nodeRole == "router" || nodeRole == "batcher" {
+			configStore, err = configstore.NewStore(conf.LocalConfig.NodeLocalConfig.FileStore.Path)
 			if err != nil {
-				return nil, nil, fmt.Errorf("could not read block %s", blockPath)
+				return nil, nil, fmt.Errorf("failed creating router config store: %s", err)
 			}
-			genesisBlock, err = protoutil.UnmarshalBlock(data)
+
+			listBlocks, err := configStore.ListBlocks()
 			if err != nil {
-				return nil, nil, fmt.Errorf("error unmarshalling to block: %s", err)
+				return nil, nil, fmt.Errorf("failed retrieve a sorted array of all block numbers: %s", err)
 			}
 
-			consensusMetaData, err := ReadSharedConfigFromBootstrapConfigBlock(genesisBlock)
-			if err != nil {
-				return nil, nil, err
+			if len(listBlocks) > 0 {
+				lastConfigBlock = listBlocks[len(listBlocks)-1]
 			}
+		}
 
-			err = proto.Unmarshal(consensusMetaData, conf.SharedConfig)
+		if lastConfigBlock == nil {
+			lastConfigBlock, err = readGenesisBlockFromBootstrapPath(conf.LocalConfig)
 			if err != nil {
-				return nil, nil, errors.Wrapf(err, "failed to unmarshal consensus metadata to a shared configuration")
+				return nil, nil, fmt.Errorf("failed to read genesis block from bootstrap file, err: %v", err)
+			}
+
+			if configStore != nil {
+				err = configStore.Add(lastConfigBlock)
+				if err != nil {
+					return nil, nil, fmt.Errorf("failed to add the genesis block to the config store: %s", err)
+				}
```

Comment:
Add logs info when it is successful, it only happens once; log the number of the last config block and whether the config store was initialized.